src/cascadia/TerminalControl/ControlInteractivity.h

```diff
@@ -0,0 +1,150 @@
+                            const uint64_t timestamp,
+                            const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                            const bool focused,
+                            const til::point terminalPosition);
+        void Touched(const winrt::Windows::Foundation::Point contactPoint);
+        void PointerMoved(const winrt::Windows::Foundation::Point mouseCursorPosition,
+                          ::Microsoft::Console::VirtualTerminal::TerminalInput::MouseButtonState buttonState,
+                          const unsigned int pointerUpdateKind,
+                          const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                          const bool focused,
+                          const til::point terminalPosition);
+        void TouchMoved(const winrt::Windows::Foundation::Point newTouchPoint,
+                        const bool focused);
+        void PointerReleased(::Microsoft::Console::VirtualTerminal::TerminalInput::MouseButtonState buttonState,
+                             const unsigned int pointerUpdateKind,
+                             const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                             const bool focused,
+                             const til::point terminalPosition);
+        void TouchReleased();
+        bool MouseWheel(const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                        const int32_t delta,
+                        const til::point terminalPosition,
+                        const ::Microsoft::Console::VirtualTerminal::TerminalInput::MouseButtonState state);
+#pragma endregion
+
+        bool CopySelectionToClipboard(bool singleLine,
+                                      const Windows::Foundation::IReference<CopyFormat>& formats);
+        void PasteTextFromClipboard();
+        void SetEndSelectionPoint(const til::point terminalPosition);
+
+    private:
+        winrt::com_ptr<ControlCore> _core{ nullptr };
+        unsigned int _rowsToScroll;
+
+        // If this is set, then we assume we are in the middle of panning the
+        //      viewport via touch input.
+        std::optional<winrt::Windows::Foundation::Point> _touchAnchor;
+
+        using Timestamp = uint64_t;
+
+        // imported from WinUser
+        // Used for PointerPoint.Timestamp Property (https://docs.microsoft.com/en-us/uwp/api/windows.ui.input.pointerpoint.timestamp#Windows_UI_Input_PointerPoint_Timestamp)
+        Timestamp _multiClickTimer;
+        unsigned int _multiClickCounter;
+        Timestamp _lastMouseClickTimestamp;
+        std::optional<winrt::Windows::Foundation::Point> _lastMouseClickPos;
+        std::optional<winrt::Windows::Foundation::Point> _singleClickTouchdownPos;
+        std::optional<til::point> _singleClickTouchdownTerminalPos;
+        std::optional<winrt::Windows::Foundation::Point> _lastMouseClickPosNoSelection;

```

Comment:
in wpf i made these use til; i will handle the merge for the click touchdown code.

Reply:
oops I accidentally the whole thing